pylint/checkers/classes.py

```diff
@@ -649,29 +681,40 @@ def visit_functiondef(self, node):
         if not node.is_method():
             return
 
-        self._check_useless_super_delegation(node)
-
         klass = node.parent.frame()
         self._meth_could_be_func = True
         # check first argument is self if this is actually a method
         self._check_first_arg_for_type(node, klass.type == 'metaclass')
         if node.name == '__init__':
+            self._check_useless_super_delegation(node)
             self._check_init(node)
             return
-        # check signature if the method overloads inherited method
-        for overridden in klass.local_attr_ancestors(node.name):
-            # get astroid for the searched method
-            try:
-                meth_node = overridden[node.name]
-            except KeyError:
-                # we have found the method but it's not in the local
-                # dictionary.
-                # This may happen with astroid build from living objects
-                continue
-            if not isinstance(meth_node, astroid.FunctionDef):
-                continue
-            self._check_signature(node, meth_node, 'overridden', klass)
-            break
+
+        # if the method overloads inherited method then
+        # - check for useless super delegation if the argument default values are
+        #   the same in the base method and in the overridden one
+        # - check signature
+        # otherwise just check for useless super delegation
+        if list(klass.local_attr_ancestors(node.name)):
```

Comment:
jsut a nitpick - make `list(klass.local_attr_ancestors(node.name))` a local variable - do method call once, not twice.